src/lib.rs

```diff
@@ -1453,4 +1456,57 @@ mod tests {
             .map(|i| u8::from_str_radix(&s[i..i + 2], 16).unwrap())
             .collect()
     }
+
+    #[test]
+    fn test_nullkey() {
+        let mut root = Node::default();
+        root = insert_leaf(&mut root, &NibbleKey::from(vec![0u8; 32]), vec![0u8; 32]).unwrap();
+        root = insert_leaf(
+            &mut root,
+            &NibbleKey::from(ByteKey::from(
+                hex::decode("11111111111111110000000000000000").unwrap(),
+            )),
+            vec![0u8; 32],
+        )
+        .unwrap();
+        let proof = make_multiproof(&root, vec![NibbleKey::from(vec![1u8; 32])]).unwrap();
+        assert_eq!(
+            proof.hashes,
+            vec![[
+                251, 145, 132, 252, 92, 249, 202, 65, 20, 16, 160, 32, 246, 163, 155, 125, 17, 186,
+                16, 171, 64, 108, 250, 70, 60, 207, 16, 164, 199, 41, 252, 143
+            ]]
+        );
+        assert_eq!(
+            proof.keyvals,
+            vec![[210, 144, 49, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17, 128]]
```

Comment:
I think the leaf with key `11111111111111110000000000000000` should be sent in `proof.keyvals`. That's the only way the verifier can make sure that the top branch node is pointing only to one leaf, which is not the leaf we're looking for.

Reply:
If I'm not mistaken, it seems for supporting null keys #25 is a requirement.

Reply:
yes, `11111111111111110000000000000000` should be present, and that's (part of) the bug.